### PR TITLE
#67 Add a Select All command.

### PR DESCRIPTION
**Pull request by MrTopom**

### CODE REVIEW

File: PS4k/resource/bundle/LangBundle.properties [hunk 1]
```diff
@@ -115,31 +117,31 @@ selection_tool_rotatez_tooltip = Rotate Around Z-Axe
...
-selection_tool_mirrorz270_caption = Rotate Z (270�)
+selection_tool_mirrorx90_caption = Rotate X (90�)
+selection_tool_mirrorx180_caption = Rotate X (180�)
+selection_tool_mirrorx270_caption = Rotate X (270�)
+selection_tool_mirrory90_caption = Rotate Y (90�)
+selection_tool_mirrory180_caption = Rotate Y (180�)
+selection_tool_mirrory270_caption = Rotate Y (270�)
+selection_tool_mirrorz90_caption = Rotate Z (90�)
+selection_tool_mirrorz180_caption = Rotate Z (180�)
+selection_tool_mirrorz270_caption = Rotate Z (270�)
```
Comment (simlu) [inline PS4k/resource/bundle/LangBundle.properties]: What's up with these. Why are they different? Is this just a Mac encoding thing?

File: PS4k/src/com/vitco/manager/action/types/SelectLogicAction.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package com.vitco.manager.action.types;
```
Comment (simlu) [inline PS4k/src/com/vitco/manager/action/types/SelectLogicAction.java]: I don't think you need to put it into a separate file. Have a look how I did it with the Camera:
https://github.com/simlu/voxelshop/blob/master/PS4k/src/com/vitco/layout/content/mainview/MainView.java#L388

Comment (MrTopom) [inline PS4k/src/com/vitco/manager/action/types/SelectLogicAction.java]: The action could be used in the plugins maybe so I thought would be better outside.

File: PS4k/src/com/vitco/layout/content/menu/SelectBarLogic.java [hunk 3]
```diff
@@ -570,5 +560,46 @@ public void onVoxelDataChanged() {
 
     }
 
+    abstract class SelectLogicAction extends StateActionPrototype {
+
+        public abstract Integer[] getVoxelsToSelect();
+
+        private Data data;
+
+        protected SelectLogicAction(Data data) {
```
Comment (simlu) [inline PS4k/src/com/vitco/layout/content/menu/SelectBarLogic.java]: Do you need a constructor and data here? I think it's already available.

File: PS4k/src/com/vitco/layout/content/menu/SelectBarLogic.java [hunk 4]
```diff
@@ -170,42 +171,31 @@ public boolean getStatus() {
         });
 
         // select all, expand selection, move to new layer, recolor
-        actionGroupManager.addAction("selection_interaction", "selection_tool_select_all", new StateActionPrototype() {
```
Comment (simlu) [inline PS4k/src/com/vitco/layout/content/menu/SelectBarLogic.java]: I would move the abstract class above this line, so it's directly where it's used.

**Comment by simlu** [comment]
This looks a lot cleaner now. Will merge tonight!

File: PS4k/src/com/vitco/layout/content/menu/SelectBarLogic.java [hunk 5]
```diff
@@ -169,43 +170,67 @@ public boolean getStatus() {
...
-        // select all, expand selection, move to new layer, recolor
-        actionGroupManager.addAction("selection_interaction", "selection_tool_select_all", new StateActionPrototype() {
+        abstract class SelectLogicAction extends StateActionPrototype {
+
+            public abstract Integer[] getVoxelsToSelect();
+
             @Override
             public void action(ActionEvent actionEvent) {
-                if (getStatus()) {
-                    // todo make this an intent (select layer) -> to prevent two history entries
```
Comment (simlu) [inline PS4k/src/com/vitco/layout/content/menu/SelectBarLogic.java]: This todo got lost, I'll re-add it after the merge.

File: PS4k/src/com/vitco/core/data/VoxelData.java [hunk 6]
```diff
@@ -22,6 +22,7 @@
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.Set;
```
Comment (simlu) [inline PS4k/src/com/vitco/core/data/VoxelData.java]: Pretty sure this is not used anymore. Always have a look at your commits, sometimes they include weirdness like this.

**Comment by simlu** [comment]
Urg and your created against master and I didn't see that lol

**Pull request merged**